iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +30,653 @@ using namespace iox::capro;
 class ServiceDescription_test : public Test
 {
   public:
-    uint16_t testAnyServiceID = iox::capro::AnyService;
-    uint16_t testAnyEventID = iox::capro::AnyEvent;
-    uint16_t testAnyInstanceID = iox::capro::AnyInstance;
-    IdString_t service1{"Service"};
-    IdString_t instance1{"Instance"};
-    IdString_t event1{"Event"};
-
-    ServiceDescription csd1 = ServiceDescription(1, 2, 3);
-    ServiceDescription csd1Eq = ServiceDescription(testAnyServiceID, 2, 3);
-    ServiceDescription csd1Ne = ServiceDescription(testAnyServiceID, 9, 3);
-    ServiceDescription csd2 = ServiceDescription(1, testAnyEventID, 3);
-    ServiceDescription csd2Eq = ServiceDescription(1, 2, 3);
-    ServiceDescription csd2Ne = ServiceDescription(1, 2, 9);
-    ServiceDescription csd3 = ServiceDescription(1, 2, testAnyInstanceID);
-    ServiceDescription csd3Eq = ServiceDescription(1, 2, testAnyInstanceID);
-    ServiceDescription csd3Ne = ServiceDescription(9, 2, testAnyInstanceID);
-    ServiceDescription csd4 = ServiceDescription(1, 2, 3);
-    ServiceDescription csd4Eq = ServiceDescription(1, 2, 3);
-    ServiceDescription csd4Ne = ServiceDescription(9, 9, 9);
-    ServiceDescription csdIdStr = ServiceDescription(service1, instance1, event1);
+    IdString_t testService;
+    IdString_t testInstance;
+    IdString_t testEvent;
     void SetUp(){};
     void TearDown(){};
 };
 
-TEST_F(ServiceDescription_test, CtorSerial)
+/// BEGIN CLASSHASH TESTS
+
+TEST_F(ServiceDescription_test, ServiceDescriptionClassHashDefaultCtorCreatesClassHashWithDefaultValues)
+{
+    ServiceDescription::ClassHash testHash{};
+
+    EXPECT_EQ(uint32_t(0), testHash[0]);
+    EXPECT_EQ(uint32_t(0), testHash[1]);
+    EXPECT_EQ(uint32_t(0), testHash[2]);
+    EXPECT_EQ(uint32_t(0), testHash[3]);
+}
+
+TEST_F(ServiceDescription_test, ServiceDescriptionClassHashCtorCreatesClassHashWithValuesPassedToTheCtor)
+{
+    ServiceDescription::ClassHash testHash{1U, 2U, 3U, 4U};
+
+    EXPECT_EQ(uint32_t(1), testHash[0]);
+    EXPECT_EQ(uint32_t(2), testHash[1]);
+    EXPECT_EQ(uint32_t(3), testHash[2]);
+    EXPECT_EQ(uint32_t(4), testHash[3]);
+}
+
+TEST_F(ServiceDescription_test, ClassHashIndexOperatorReturnsTheValueInTheCorrespondingIndex)
```

Comment:
This test is identical to the previous when, the only difference is that you use once EXPECT_EQ and once EXPECT_THAT.